code/datums/components/thermite.dm

```diff
@@ -1,76 +1,109 @@
 		/turf/open/lava,
 		/turf/open/space,
 		/turf/open/water,
 		/turf/open/chasm,
 	))
-	///List of turfs that are immune to thermite
+	/// List of turfs that are immune to thermite
 	var/static/list/immunelist = typecacheof(list(
 		/turf/closed/wall/mineral/diamond,
 		/turf/closed/indestructible,
 		/turf/open/indestructible,
 	))
-	///List of turfs that take extra thermite to burn through
+	/// List of turfs that take extra thermite to burn through
 	var/static/list/resistlist = typecacheof(list(
 		/turf/closed/wall/r_wall,
 	))
 
-/datum/component/thermite/Initialize(_amount)
-	if(!istype(parent, /turf) || blacklist[parent.type])
+/datum/component/thermite/Initialize(amount = 50, thermite_overlay = default_thermite_overlay)
+	if(!isturf(parent))
 		return COMPONENT_INCOMPATIBLE
+	//not actually incompatible, but not valid
+	if(blacklist[parent.type])
+		qdel(src)
+		return
 
 	if(immunelist[parent.type])
-		amount = 0 //Yeah the overlay can still go on it and be cleaned but you arent burning down a diamond wall
+		src.amount = 0 //Yeah the overlay can still go on it and be cleaned but you arent burning down a diamond wall
 	else
-		amount = _amount
+		src.amount = amount
 		if(resistlist[parent.type])
 			burn_require = 50
 		else
 			burn_require = 30
 
-	var/turf/master = parent
-	overlay = mutable_appearance('icons/effects/effects.dmi', "thermite")
-	master.add_overlay(overlay)
+	src.thermite_overlay = thermite_overlay
 
+/datum/component/thermite/Destroy()
+	thermite_overlay = null
+	if(burn_callback)
+		QDEL_NULL(burn_callback)
+	if(burn_timer)
+		deltimer(burn_timer)
+		burn_timer = null
+	if(fakefire)
+		QDEL_NULL(fakefire)
+	return ..()
+
+/datum/component/thermite/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, PROC_REF(on_examine))
+	RegisterSignal(parent, COMSIG_ATOM_UPDATE_OVERLAYS, PROC_REF(on_update_overlays))
 	RegisterSignal(parent, COMSIG_COMPONENT_CLEAN_ACT, PROC_REF(clean_react))
 	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, PROC_REF(attackby_react))
 	RegisterSignal(parent, COMSIG_ATOM_FIRE_ACT, PROC_REF(flame_react))
+	RegisterSignal(parent, COMSIG_PARENT_QDELETING, PROC_REF(parent_qdeleting)) //probably necessary because turfs are wack
+	var/turf/turf_parent = parent
+	turf_parent.update_appearance()
 
 /datum/component/thermite/UnregisterFromParent()
-	UnregisterSignal(parent, COMSIG_COMPONENT_CLEAN_ACT)
-	UnregisterSignal(parent, COMSIG_PARENT_ATTACKBY)
-	UnregisterSignal(parent, COMSIG_ATOM_FIRE_ACT)
-	UnregisterSignal(parent, COMSIG_PARENT_QDELETING)
-
-/datum/component/thermite/Destroy()
-	var/turf/master = parent
-	master.cut_overlay(overlay)
-	return ..()
-
-/datum/component/thermite/InheritComponent(datum/component/thermite/newC, i_am_original, _amount)
+	UnregisterSignal(parent, list(
+		COMSIG_PARENT_EXAMINE,
+		COMSIG_ATOM_UPDATE_OVERLAYS,
+		COMSIG_COMPONENT_CLEAN_ACT,
+		COMSIG_PARENT_ATTACKBY,
+		COMSIG_ATOM_FIRE_ACT,
+		COMSIG_PARENT_QDELETING))
```

Comment:
```suggestion
		COMSIG_PARENT_QDELETING,
	))
```

may as well multiline it correctly. also this can be alphabetized, right? if not it's okay

Reply:
eh i guess it can